pkg/workceptor/kubernetes.go

```diff
@@ -374,7 +374,23 @@ mainLoop:
 		for { // check between every line read to see if we need to stop reading
 			line, err := streamReader.ReadString('\n')
 			if err != nil {
-				// First check if the error is not EOF, if error is not EOF retry 5 times if error persists set error and mark the job as failed.
+				// Check if the context was canceled and the work state isn't "Succeeded".
+				// If so, set the error and mark the job as failed.
+				if kw.GetContext().Err() == context.Canceled {
+					if kw.Status().State != WorkStateSucceeded {
```

Comment:
The context could also be canceled for both a failed and canceled job, 
I think we add an or for both these cases also